framework/api/routes/web_scraper/web_scraper_router.py

```diff
@@ -17,6 +18,7 @@
 # TODO: Possibly just belongs on the product router?? why make this distinction (feels bad tbh, shows implementation details)
 # TODO Should use DI once web scraper is setup more clean, web scraper is a hosted service under the API
 @WEB_SCRAPER_ROUTER.route("/search", methods=["POST"])
+@request_object('create_product_async', SearchForProductQuery)
```

Comment:
so from what i'm reading this is some sort of decorator. How does it or is it actually working here/

Reply:
yeah so in python you can make decorators (this one lives in the middleware file, may not be an appropriate place for it to live. You can create decorators in python to add additional functionality to a function. I believe they get called when the function itself gets called...which makes me wonder now if each endpoint call re-registers the request object...hmm...but yeah, with a decorator you don't have to modify the function in any way.

Reply:
Makes sense, ty.

Uuuh idk which how to underlying decorator works, i'll leave this comment for you to resolve incase it is an issue you want to address.

(If github allows you to resolve ny comment)